pallets/allocations/src/migrations.rs

```diff
@@ -0,0 +1,162 @@
+			let current = Pallet::<T>::current_storage_version();
+			let onchain = Pallet::<T>::on_chain_storage_version();
+
+			log::info!(
+				"on_runtime_upgrade[{:#?}]=> Running migration with current storage version {:?} / onchain {:?}",
+				line!(),
+				current,
+				onchain
+			);
+
+			if current == 1 && onchain == 0 {
+				let pallet_prefix: &[u8] = b"Allocations";
+				let storage_item_prefix: &[u8] = b"Oracles";
+
+				if have_storage_value(pallet_prefix, storage_item_prefix, &[]) {
+					remove_storage_prefix(pallet_prefix, storage_item_prefix, &[]);
+
+					current.put::<Pallet<T>>();
+
+					log::info!(
+						"on_runtime_upgrade[{:#?}]=> Removed Oracles, Migrated to storage version {:?}",
+						line!(),
+						current
+					);
+				} else {
+					log::error!("on_runtime_upgrade[{:#?}]=> Oracles doesn't exist", line!(),);
+				}
+
+				T::DbWeight::get().reads_writes(1, 1)
+			} else {
+				log::info!(
+					"on_runtime_upgrade[{:#?}]=> Migration did not executed. This probably should be removed",
+					line!(),
+				);
+				T::DbWeight::get().reads(1)
+			}
+		}
+
+		#[cfg(feature = "try-runtime")]
+		fn pre_upgrade() -> Result<(), &'static str> {
+			use frame_support::traits::OnRuntimeUpgradeHelpersExt;
+
+			let current = Pallet::<T>::current_storage_version();
+			let onchain = Pallet::<T>::on_chain_storage_version();
+
+			log::info!(
+				"pre_upgrade[{:#?}]=> with current storage version {:?} / onchain {:?}",
+				line!(),
+				current,
+				onchain
+			);
+
+			if current == 1 && onchain == 0 {
+				let pallet_prefix: &[u8] = b"Allocations";
+				let storage_item_prefix: &[u8] = b"Oracles";
+
+				let maybe_stored_data = get_storage_value::<Vec<T::AccountId>>(pallet_prefix, storage_item_prefix, &[]);
+
+				if let Some(stored_data) = maybe_stored_data {
+					Self::set_temp_storage(stored_data.len() as u32, "oracles_count");
+					log::info!(
+						"pre_upgrade[{:#?}]=> Oracles count :: [{:#?}]",
+						line!(),
+						stored_data.len(),
+					);
+				} else {
+					log::info!(
+						"pre_upgrade[{:#?}]=> Oracles is_none :: [{:#?}]",
+						line!(),
+						maybe_stored_data.is_none(),
+					);
+				}
+			} else {
+				log::info!(
+					"pre_upgrade[{:#?}]=> Version mismatch. This probably should be removed",
+					line!(),
+				);
+			}
+
+			Ok(())
+		}
+
+		#[cfg(feature = "try-runtime")]
+		fn post_upgrade() -> Result<(), &'static str> {
+			use frame_support::traits::OnRuntimeUpgradeHelpersExt;
+
+			let current = Pallet::<T>::current_storage_version();
+			let onchain = Pallet::<T>::on_chain_storage_version();
+
+			log::info!(
+				"post_upgrade[{:#?}]=> with current storage version {:?} / onchain {:?}",
+				line!(),
+				current,
+				onchain
+			);
+
+			if current == 1 && onchain == 1 {
+				let pallet_prefix: &[u8] = b"Allocations";
+				let storage_item_prefix: &[u8] = b"Oracles";
+
+				if get_storage_value::<Vec<T::AccountId>>(pallet_prefix, storage_item_prefix, &[]).is_none() {
```

Comment:
Post migration you cannot do this test as `Oracles` is removed, isn't it? In that case please feel free to remove this check and no need to store the pre-migration numbers in pre-upgrade either

Reply:
Here we are ensuring storage `Oracles` is removed. I hope better to ensure that state. As suggested removed the temp storage of Oracles count.